generator/views/funnel_analysis_view.py

```diff
@@ -0,0 +1,203 @@
+"""Class to describe a Funnel Analysis View."""
+from __future__ import annotations
+
+from typing import Any, Dict, Iterator, List, Optional
+
+from .view import View, ViewDict
+
+
+class FunnelAnalysisView(View):
+    """A view for Client Counting measures."""
+
+    type: str = "funnel_analysis_view"
+    num_funnel_steps: int = 4
+
+    def __init__(self, namespace: str, tables: List[Dict[str, str]]):
+        """Get an instance of a FunnelAnalysisView."""
+        super().__init__(namespace, "funnel_analysis", FunnelAnalysisView.type, tables)
+
+    @classmethod
+    def from_db_views(
+        klass,
+        namespace: str,
+        is_glean: bool,
+        channels: List[Dict[str, str]],
+        db_views: dict,
+        num_funnel_steps: int = num_funnel_steps,
+    ) -> Iterator[FunnelAnalysisView]:
+        """Get Client Count Views from db views and app variants."""
+        # We can guarantee there will always be at least one channel,
+        # because this comes from the associated _get_glean_repos in
+        # namespaces.py
+        dataset = next(
+            (channel for channel in channels if channel.get("channel") == "release"),
+            channels[0],
+        )["dataset"]
+
+        necessary_views = {"events_daily", "event_types"}
+        actual_views = {}
+        for view_id, references in db_views[dataset].items():
+            if view_id in necessary_views:
+                actual_views[view_id] = f"`mozdata.{dataset}.{view_id}`"
+
+        if len(actual_views) == 2:
```

Comment:
So if this isn't true it yields nothing? Some comments to explain the behaviour here might be good.

Reply:
Right, we only yield a funnel analysis view if we have all the underlying tables. Will add a comment.